src/plugins/data/public/antlr/opensearch_sql/code_completion.ts

```diff
@@ -47,80 +48,59 @@ export const getSuggestions = async ({
 }: QuerySuggestionGetFnArgs): Promise<QuerySuggestion[]> => {
   const { api } = services.uiSettings;
   const dataSetManager = services.data.query.dataSetManager;
+  const { lineNumber, column } = position || {};
   const suggestions = getOpenSearchSqlAutoCompleteSuggestions(query, {
-    line: position?.lineNumber || selectionStart,
-    column: position?.column || selectionEnd,
+    line: lineNumber || selectionStart,
+    column: column || selectionEnd,
   });
 
-  const finalSuggestions = [];
+  const finalSuggestions: QuerySuggestion[] = [];
 
   try {
     // Fetch columns and values
-    if ('suggestColumns' in suggestions && (suggestions.suggestColumns?.tables?.length ?? 0) > 0) {
-      const tableNames = suggestions.suggestColumns?.tables?.map((table) => table.name) ?? [];
-      const schemas = await fetchTableSchemas(tableNames, api, services);
-
-      schemas.forEach((schema) => {
-        if (schema.body?.fields?.length > 0) {
-          const columns = schema.body.fields.find((col: any) => col.name === 'COLUMN_NAME');
-          const fieldTypes = schema.body.fields.find((col: any) => col.name === 'DATA_TYPE');
+    if (suggestions.suggestColumns?.tables?.length) {
+      const tableNames = suggestions.suggestColumns.tables.map((table) => table.name);
+      const schemas = await fetchTableSchemas(tableNames, api, dataSetManager);
+
+      (schemas as IDataFrameResponse[]).forEach((schema: IDataFrameResponse) => {
```

Comment:
nit. it would be better to define return types for `fetchTableSchemas` instead of assert